### ISSUE TITLE
VShaxe stalled

### ISSUE THREAD

**Issue by ncannasse** [vshaxe/vshaxe#389]
One of our devs got several times VSHaxe stalled today.
Behavior : 
- VSCode stuck in Building... upon compilation , because some task is not yet complete
- VSCode is responsive, two haxe.exe processes at 0% CPU each
- One Code.exe (vshaxe ?) seems to be in some kind of loop (eats 8% of CPU at constant rate)

Killing this process gives the attached log here : [VSLog.txt](https://github.com/vshaxe/vshaxe/files/3794323/VSLog.txt)
Please note all the pending requests gets displayed as error after the kill.

Don't hesitate to tell me if/how I can gather more information

**Comment by Gama11** [comment]
I'm not sure how I would debug this myself. Probably build vshaxe from source and attach a debugger to the language server (and the "pause" and see what the callstack looks like? Not sure if that would work), but that kind of requires a somewhat consistent way to reproduce.

That log file is not very helpful unfortunately, basically it just says that the language server has crashed (because you killed it I guess).

**Comment by ncannasse** [comment]
Yes, I think it also says that it was processing a hover request when it freezed ? 
Maybe looking at some `while` in the code and ensuring that they always terminate ?

**Comment by Gama11** [comment]
I think you already suggested that once before, and even searched through the code and found that there's barely any while loops. ;)
